config/k8s-installers/capi/wc-config.yaml

```
@@ -0,0 +1,23 @@
+hnc:
+  excludedNamespaces:
+    - calico-apiserver
+    - calico-system
+    - alertmanager
+    - cert-manager
+    - falco
+    - fluentd
+    - fluentd-system
+    - hnc-system
+    - ingress-nginx
+    - kube-node-lease
+    - kube-public
+    - kube-system
+    - kured
+    - monitoring
+    - rook-ceph
+    - velero
+    - openstack-system
+    - gatekeeper-system
+    - gpu-operator
+    - metallb-system
+    - tigera-operator
```

Could you try only adding the capi namespaces here and see if it merges correctly with default list? :slightly_smiling_face: 
```suggestion
    - openstack-system
    - gatekeeper-system
    - gpu-operator
    - metallb-system
    - tigera-operator
```

---

That was the first thing I tried and that did not work :(

---

Hmm, I dislike the fact that this is now duplicating information. But I understand that the other way didn't work.

Suggestion: Do we need to have this in the config? Could we perhaps instead move it to the values file and have an if-statement that adds the capi namespaces if capi=true? Then just use this config if we want to add something extra.

---

Thanks, I like this solution better.

Suggestion: Could we perhaps move [this entire list](https://github.com/elastisys/compliantkubernetes-apps/blob/main/config/wc-config.yaml#L124-L140) to the same place? And just have the config option for extra namespaces that we might want to exclude. What do you think of that?

---

Hmm, I realize now that we also have the `excludedExtraNamespaces` config option. If we were to move the list of "standard" exclusions to the values file. Then perhaps we do not need both `excludedExtraNamespaces` and `excludedNamespaces`. I wonder if there are cases where we would rather have this in config so that it is possible to change for specific environments.

---

To me, it seems extremely unlikely that we would want to configure any of these namespaces to be hnc-managed. So to me it should suffice with only having the option to add namespaces with the `excludedExtraNamespaces`.

---

Maybe it is a bit out of scope for the original task. But I wonder if we should implement that change.
Although perhaps we should check with others before going ahead, to see if it is a good idea.
I'll @ the ones that have looked at this PR at least
@simonklb @Xartos @OlleLarsson @robinAwallace

---

I'm always for less config and more streamlined defaults. :)

---

Let me know if I follow this correctly, you are suggesting moving [this](https://github.com/elastisys/compliantkubernetes-apps/blob/main/config/wc-config.yaml#L124) to [here](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile.d/values/hnc/controller.yaml.gotmpl#L3)?

---

@OlleLarsson yes, and removing the config option from your first link since we have another config option that can be used to exclude namespaces.

---

Then I think it would make sense to also add this to public documentation, that these are "reserved" namespaces that the customer are not allowed to create. I'm mainly thinking of the namespaces `rook-ceph` and `metallb-system` that might not be part of the cluster, and also some kind of safeguard for customers to create these namespaces since they will not get the correct labels and stuff.

---

Do we have no idea if the cluster is using Rook or MetalLB in Apps? I'm thinking that it would be fairly easy to make them conditional if we did (note I'm not asking for it to be introduced if it isn't already).

---

If I remember correctly we have that for rook, but not metallb

---

Wouldn't it make sense to not conditional any namespaces based on k8s-installer or add-ons? That would make the transition of moving environments between k8s-installers or add-ons "easier"?

---

Yea that is true, good thinking! While highly unlikely, a customer creating a namespace `metallb-system` would make it problematic for us to in the future start using that namespace. :smile: Let's just reserve all of them.